session.go

```diff
@@ -170,19 +170,22 @@ func (s *Session) Send(r *Request) (response *Response, err error) {
 	//
 
 	// Debug log request
-	s.log("--------------------------------------------------------------------------------")
-	s.log("REQUEST")
-	s.log("--------------------------------------------------------------------------------")
-	s.log("Method:", req.Method)
-	s.log("URL:", req.URL)
-	s.log("Header:", req.Header)
-	s.log("Form:", req.Form)
-	s.log("Payload:")
-	if r.RawPayload && s.Log && buf != nil {
-		s.log(base64.StdEncoding.EncodeToString(buf.Bytes()))
-	} else {
-		s.log(pretty(r.Payload))
+	if s.Log {
+		s.log("--------------------------------------------------------------------------------")
+		s.log("REQUEST")
+		s.log("--------------------------------------------------------------------------------")
+		s.log("Method:", req.Method)
+		s.log("URL:", req.URL)
+		s.log("Header:", req.Header)
+		s.log("Form:", req.Form)
+		s.log("Payload:")
+		if r.RawPayload && s.Log && buf != nil {
```

Comment:
Yes. When logging is disabled, it now checks once for this whole block instead of many times, and it avoids the expensive pretty() or the base64 encoding parts.  In other words it optimised for the case when logging is off and I don't worry about checking twice when logging is on.